File: Baekjoon/Baekjoon04963.cpp
// Baekjoon04963.cpp
#include <iostream>

#include <algorithm>

using namespace std;

int moves[3] = { -1, 0, 1 };

int w, h;
int map[50][50];
bool visited[50][50];

void dfs(int row, int col) {
    for (int i = 0; i < 3; i++) {
        for (int j = 0; j < 3; j++) {
            int newRow = row + moves[i], newCol = col + moves[j];
            if ((0 <= newRow && newRow < h) && (0 <= newCol && newCol < w)) {
                if (!visited[newRow][newCol] && map[newRow][newCol] == 1) {
                    visited[newRow][newCol] = true;
                    dfs(newRow, newCol);
                }
            }
        }
    }
}

void solve() {
    for (int i = 0; i < h; i++) {
        for (int j = 0; j < w; j++)
            cin >> map[i][j];
        fill(visited[i], visited[i] + w, false);
    }

    int ans = 0;
    for (int i = 0; i < h; i++) {
        for (int j = 0; j < w; j++) {
            if (!visited[i][j]) {
                visited[i][j] = true;
                if (map[i][j] == 1) {
                    dfs(i, j);
                    ans++;
                }
            }
        }
    }

    cout << ans << '\n';
}

int main(void) {
    cin >> w >> h;

    while (!(w == 0 && h == 0)) {
        solve();
        cin >> w >> h;
    }

    return 0;
}

File: Baekjoon/Baekjoon04963.py
# Baekjoon04963.py
import sys
input = sys.stdin.readline

MOVES = [(-1, -1), (-1, 0), (-1, 1), (0, -1), (0, 1), (1, -1), (1, 0), (1, 1)]

def dfs(row, col):
    for r, c in MOVES:
        nextRow, nextCol = row + r, col + c

        if (0 <= nextRow < maxRow) and (0 <= nextCol < maxCol):
            if not visited[nextRow][nextCol] and adjMat[row][col]:
                visited[nextRow][nextCol] = True

                dfs(nextRow, nextCol)

maxCol, maxRow = map(int, input().split())
while (maxCol != 0 and maxRow != 0):
    ans = 0
    visited = [[False] * maxCol for _ in range(maxRow)]

    adjMat = [list(map(int, input().split())) for _ in range(maxRow)]

    for row in range(maxRow):
        for col in range(maxCol):
            if not visited[row][col] and adjMat[row][col]:
                visited[row][col] = True
                dfs(row, col)

                ans += 1

    print(ans)

    maxCol, maxRow = map(int, input().split())
